### ISSUE TITLE
vectorAction clipped

### ISSUE THREAD

**Issue by Hallam-Liu** [Unity-Technologies/ml-agents#2416]

**Comment by Hallam-Liu** [comment]
What if I want the vectioaction's range larger than[1,1]

**Comment by arixlin** [comment]
@Hallam-Liu 
` Mathf.Clamp(vectorAction[3], -2f, 2f);`

**Comment by ervteng** [comment]
Also - no reason why you can't multiply the vectorAction by some value in AgentAction. However, the neural network will only output values between [-1,1].

**Comment by Hallam-Liu** [comment]
> Also - no reason why you can't multiply the vectorAction by some value in AgentAction. However, the neural network will only output values between [-1,1].

Yeah，But I want the output of model larger than [-1,1]. Can I do it in ml agent?

**Comment by Hallam-Liu** [comment]
> @Hallam-Liu
> ` Mathf.Clamp(vectorAction[3], -2f, 2f);`

Then all the out put will be -2 or 2 if the output of model larger than 1 or smaller than -1

**Comment by Hallam-Liu** [comment]
> Also - no reason why you can't multiply the vectorAction by some value in AgentAction. However, the neural network will only output values between [-1,1].

The problem is I have decisions which spacesize is 120 in Heuristic brain. and most of their values larger than 1. And I want to imitate the Heuristic brain but cannot get output larger than 1. It is like a supervised learning and let the 120 output work like as labels

**Comment by ervteng** [comment]
Recommendation would be to scale down the Heuristic brain output to [-1,1], then in game code (DecideAction) multiply that by the opposite of the scale. That way you'll have the same output space in both brains.

**Comment by Hallam-Liu** [comment]
> Recommendation would be to scale down the Heuristic brain output to [-1,1], then in game code (DecideAction) multiply that by the opposite of the scale. That way you'll have the same output space in both brains.

Actually I tried but the loss descent very fast lower than 0.1. It is hard to decide whether the training is converge or not since all the outputs scaled down to [-1,1]

**Comment by awjuliani** [comment]
Hi all. Closing this issue due to inactivity. Feel free to re-open if you feel necessary.
